profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -1,142 +1,151 @@
 
-  return (<div>
-    {showButton()}
-  </div>)
+export function FunctionsAsComponents({ buttonText = "Start Now" }) {
+  return <button>{buttonText}</button>;
 }
 
+// shallow copy
 export function objectCopying() {
-  const object = { a: { c: 1 }, b: 2 }
-  const copy = { ...object }
-  copy.a.c = 2
-  return { ...object }
+  const object = { a: { c: 1 }, b: 2 };
+  // What is this supposed to do? Not returning the changed object
+  const copy = { ...object };
+  copy.a.c = 2;
+  return {
+    ...object,
+    a: {
+      ...object.a,
+      c: 2,
+    },
+  };
 }
 
 export function arrayCopying() {
-  const array = [{ a: 1}, {a: 2}, {a: 3}]
-  const copy = [...array]
-  return copy;
+  const array = [{ a: 1 }, { a: 2 }, { a: 3 }];
+  return [...array];
 }
 
-// user abort
-export function UseEffect({ fetchURL, label }) {
+function useFetch(url) {
+  const [response, setResponse] = useState({});
   useEffect(() => {
+    setResponse({ isLoading: true });
     const fetchData = async () => {
-      await fetch(fetchURL);
+      try {
+        const data = await fetch(url).then((blob) => blob.json());
+        setResponse({ isLoading: false, data });
+      } catch (error) {
+        setResponse({ error, isLoading: false });
+      }
     };
-    
+
     fetchData();
-  }, [fetchURL]);
+  }, [url]);
+
+  return response;
+}
+
+// user abort
+export function UseEffect({ fetchURL, label }) {
+  const { isLoading, data, error } = useFetch(fetchURL);
 
+  // do something with this.
   return (
     <div>
+      {isLoading ? <p>Loading...</p> : null}
+      {data ? <pre>{JSON.stringify(data, null, 2)}</pre> : null}
+      {error ? <p>{JSON.stringify(error, null, 2)}</p> : null}
       <button>{label}</button>
     </div>
   );
 }
 
 export function UseEffectDerivedCalculation() {
-  const [remainder, setReminder] = useState()
-  const [clickedTimes, setClickedTimes] = useState()
+  const [clickedTimes, setClickedTimes] = useState(0);
 
-  useEffect(() => {
-    setReminder(clickedTimes % 5)
-  }, [clickedTimes])
-
-  const handleClick = () => setClickedTimes(clickedTimes + 1)
+  const handleClick = () => setClickedTimes(clickedTimes + 1);
 
   return (
     <div>
       <button onClick={handleClick}>Add Click Count</button>
-      <span>
-        {sum}
-      </span>
-      <span>
-        {remainder}
-      </span>
+      {/* <span>
+         {sum} variable does not exist, remove
+       </span>*/}
+      <span>{clickedTimes % 5}</span>
     </div>
-  )
+  );
 }
 
 export function UseEffectLifeCycle() {
-  const [_loaded, setLoaded] = useState()
+  const [loaded, setLoaded] = useState(false);
 
   useEffect(() => {
-    setTimeout(() => setLoaded(true), 1000)
-  }, [])
+    const timeout = setTimeout(() => setLoaded(true), 1000);
 
-  const handleClick = () => setClickedTimes(clickedTimes + 1)
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, []);
 
   return (
     <div>
-      <button onClick={handleClick}>Add Click Count</button>
```

Comment:
Is this supposed to be removed? Since there's no clicked times, assumed it was copied from the exercise before